packages/plugin-hardhat/src/upgrade-beacon.ts

```diff
@@ -27,7 +27,9 @@ export function makeUpgradeBeacon(hre: HardhatRuntimeEnvironment, platformModule
 
     const UpgradeableBeaconFactory = await getUpgradeableBeaconFactory(hre, getSigner(ImplFactory.runner));
     const beaconContract = attach(UpgradeableBeaconFactory, beaconAddress);
-    const upgradeTx = await beaconContract.upgradeTo(nextImpl);
+
+    const overrides = opts.txOverrides ? [opts.txOverrides] : [];
+    const upgradeTx = await beaconContract.upgradeTo(nextImpl, ...overrides);
```

Comment:
What happens if you just write `upgradeTo(nextImpl, overrides)`?

Reply:
That would cause an error because then `overrides` appears as an array and Ethers tries to use it as part of the actual function arguments.

We need to spread the array because its element `opts.txOverrides` is the actual overrides object, and if that is not present, then it will be excluded.

Reply:
Sorry my question was actually if we write `upgradeTo(nextImpl, opts.txOverrides)` :slightly_smiling_face:

Reply:
That doesn't work, but `upgradeTo(nextImpl, opts.txOverrides ?? {})` works.  Let's use this.

Reply:
I'm not sure that these options will work with Truffle, I recall it working weird if you pass an empty object.

Reply:
Right, these don't work for Truffle.  Will leave them as-is for both Hardhat and Truffle, for consistency.